smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatManager.java

```diff
@@ -254,6 +264,44 @@ public void run() {
         });
     }
 
+    private void addXEP0249Listener(XMPPConnection connection) {
+        // Listens for all messages that include a GroupChatInvitation extension and fire the invitation
+        // listeners
+        StanzaListener xep0249InvitationPacketListener = new StanzaListener() {
+            @Override
+            public void processStanza(Stanza packet) {
+                final Message message = (Message) packet;
+                ExtensionElement inviteExt =
+                    packet.getExtension(GroupChatInvitation.class);
+                GroupChatInvitation invite =
+                    inviteExt instanceof GroupChatInvitation ? (GroupChatInvitation) inviteExt : null;
+
+                if (invite != null && invite.getRoomAddress() != null) {
```

Comment:
Those checks are not needed. The `invite.getRoomAddress() != null` check is unnecessary,because the provider will not create GroupChatInviation instances without a room address, because the according element is required to have a 'jid' attribute.